resources/views/pages/products.ejs

```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <% include ../partials/head %>
+    <link rel="stylesheet" href="/resources/css/products.css" />
+  </head>
+  <body class="container">
+    <header>
+      <% include ../partials/header %>
+    </header>
+    <main class="flex-row">
+      <div class="category-list">
+        <% for(var i=0; i < categories.length; i++) { %>
+        <a href="/products/<%= categories[i].key %>"><h1 class="category-list__name"><%= categories[i].name %></h1></a>
+        <% } %>
+      </div>
+      <div class="product-list">
+        <div class="mob-category-list-container">
+        <select class="mob-category-list" onchange="window.location = this.value;">
+        <% for(var i=0; i < categories.length; i++) { %>
+            <% if (categoryId && categoryId===categories[i].id) { %>
+                <option value="/products/<%= categories[i].key %>" selected>
+            <% } else { %>
+                <option value="/products/<%= categories[i].key %>">
+            <% } %>
+            <%= categories[i].name %>
+        </option>
+        <% } %>
+        </select>
+        </div>
+        <% for(var i=0; i < products.length;i) { %>
+        <div class="product-list-row">
+          <% for(i && i++; i < products.length; i++) { %>
+          <div class="product-card">
+            <h1 class="product-card__name"><%= products[i].name %></h1>
+            <div class="product-card__detail">
+              <img
+                class="product-card__detail-img"
+                src="<%= products[i].imageURL %>"
+                alt="<%= products[i].name %> logo"
+              />
+              <div class="product-cart__detail-text">
+                <p class="product-card__detail-description"><%= products[i].description %></p>
+                <a class="product-buy-btn product-card__detail-btn" id="<%= products[i].id %>">
+                    Buy Now @Rs.<%= products[i].price %>
+                </a>
+              </div>
+            </div>
+            <div class="product-card__buy">
+              <div class="product-card__buy-price">MRP Rs.<%= products[i].price %></div>
+              <a class="product-buy-btn product-card__buy-btn" id="<%= products[i].id %>">
+                Buy Now <span class="product-card__buy-sm-txt">@Rs.<%= products[i].price %></span>
+              </a>
+            </div>
+          </div>
+          <% if ((i+1) % 4 === 0) { break; } } %>
```

Comment:
Don't use break and continue statements. They create complex logics which are hard to maintain, and difficult to debug.

Reply:
done